src/Microsoft.DotNet.Cli.Utils/FrameworkDependencyFile.cs

```diff
@@ -45,6 +46,77 @@ public string GetNetStandardLibraryVersion()
                 ?.Version;
         }
 
+        public bool TryGetMostFitRuntimeIdentifier(
+            string alternativeCurrentRuntimeIdentifier,
+            string[] candidateRuntimeIdentifiers,
+            out string mostFitRuntimeIdentifier)
+        {
+            return TryGetMostFitRuntimeIdentifier(
+                RuntimeEnvironment.GetRuntimeIdentifier(),
+                alternativeCurrentRuntimeIdentifier,
+                DependencyContext.RuntimeGraph,
+                candidateRuntimeIdentifiers,
+                out mostFitRuntimeIdentifier);
+        }
+
+        internal static bool TryGetMostFitRuntimeIdentifier(
+            string currentRuntimeIdentifier,
+            string alternativeCurrentRuntimeIdentifier,
+            IReadOnlyList<RuntimeFallbacks> runtimeGraph,
+            string[] candidateRuntimeIdentifiers,
+            out string mostFitRuntimeIdentifier)
+        {
+            mostFitRuntimeIdentifier = null;
+            RuntimeFallbacks[] runtimeFallbacksCandidates;
+
+            if (!string.IsNullOrEmpty(currentRuntimeIdentifier))
+            {
+                runtimeFallbacksCandidates =
+                    runtimeGraph
+                    .Where(g => string.Equals(g.Runtime, currentRuntimeIdentifier, StringComparison.OrdinalIgnoreCase))
+                    .ToArray();
```

Comment:
Do we need to ToArray().  Could just let it materialize when it gets added to the List later. Less copying.

Reply:
That a different collection.

We need to find the Single RuntimeFallbacks(which is a confusing name) first and then construct the fallback.

RuntimeFallbacks has fallbacks and runtime field. 

runtimeFallbacksCandidates need Any() and Singe(), i don't want to iterate twice